### PR TITLE
Add missing dependency

### PR DESCRIPTION
**Pull request by Ayc0**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Add `@types/unist` in dependencies (in the deps and not devDeps, because those types are exposed to users)

Related to https://github.com/syntax-tree/unist-util-select/issues/12

### CODE REVIEW

**Comment by Ayc0** [comment]
@wooorm, I didn't know which version to choose, so I just picked the latest

**Comment by Ayc0** [comment]
I could also use peer deps instead of deps, WDYT?

File: package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "index.js"
   ],
   "dependencies": {
+    "@types/unist": "^2.0.6",
```
Comment (ChristianMurphy) [inline package.json]:
```suggestion
    "@types/unist": "^2.0.0",
```

**Review by ChristianMurphy — APPROVED** [review]
Thanks @Ayc0!
The rule of thumb for picking versions in unified.
Generally go for the first release of the latest major and use `^` to allow for newer releases in the same line.

**Comment by Ayc0** [comment]
Thanks @ChristianMurphy 

And what about deps vs peerDeps?
Any preferences?

**Comment by ChristianMurphy** [comment]
I'd lean `depedencies`.
`peerDependencies` should be used sparingly, and specifically where either: a user needs to be able to select another version of a package, _or_ the location of a package's folder relative to another package is important.
Here the relative folder isn't important, and selecting a specific version of the dependency manually is unlikely and could cause issues.

**Review by wooorm — APPROVED** [review]

**Comment by Ayc0** [comment]
Thanks!

**Comment by wooorm** [comment]
Thank you Benjamin!

**Pull request merged**